Descent3/aistruct.h

```diff
@@ -713,17 +715,17 @@ struct ain_weapon_hit_info {
 
 #define AI_MEM_DEPTH 5
 
-struct ai_mem {
+struct [[gnu::packed]] ai_mem {
```

Comment:
why is struct packing needed? without context, this seems smelly

Reply:
Also [[gnu::packed]] isn't portable, MSVC will just ignore the unknown attribute with [warning C5030: attribute [[gnu::packed]] is not recognized](https://github.com/DescentDevelopers/Descent3/pull/412/checks#step:6:520).
I believe `#pragma pack` is supported by all recent compiler versions nowadays.

Reply:
@tophyr this was part of [something else I was working on. ](https://github.com/DescentDevelopers/Descent3/issues/279)
@pzychotic MSVC is perpetually behind both GCC and clang. You would be better off using clang/llvm on windows.

Reply:
It isn't about what I use, it's about what this project does. I linked to the current CI build log, because we do use MSVC here and that attribute is being ignored in our windows builds. Unless this projects decides to be GCC/Clang only, this issue remains.
Also don't try to tell me which build environment I have to use, just because you don't like it.

Reply:
not really a problem because it's being removed.